File: app/src/main/java/at/shockbytes/dante/ui/fragment/dialogs/QueryDialogFragment.java
package at.shockbytes.dante.ui.fragment.dialogs;

import android.app.Dialog;
import android.content.DialogInterface;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.v4.app.DialogFragment;
import android.support.v7.app.AlertDialog;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.EditText;

import at.shockbytes.dante.R;

/**
 * @author Martin Macheiner
 *         Date: 30.08.2016.
 */
public class QueryDialogFragment extends DialogFragment {

    public interface OnQueryEnteredListener {

        void onQueryEntered(String isbn);
    }

    private EditText editQuery;

    private OnQueryEnteredListener queryListener;

    public static QueryDialogFragment newInstance() {
        return new QueryDialogFragment();
    }

    @NonNull
    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {

        AlertDialog.Builder builder = new AlertDialog.Builder(getContext());
        builder.setTitle(R.string.dialogfragment_query_title);
        builder.setIcon(R.mipmap.ic_launcher);
        builder.setMessage(R.string.dialogfragment_query_message);
        builder.setView(buildView());
        builder.setNegativeButton(android.R.string.cancel, new DialogInterface.OnClickListener() {
            @Override
            public void onClick(DialogInterface dialogInterface, int i) {
                dismiss();
            }
        });
        builder.setPositiveButton(android.R.string.search_go, new DialogInterface.OnClickListener() {
            @Override
            public void onClick(DialogInterface dialogInterface, int i) {
                if (queryListener != null) {
                    // Remove blanks with + so query works also for titles
                    queryListener.onQueryEntered(editQuery.getText().toString().replace(' ', '+'));
                }
            }
        });

        return builder.create();
    }


    public void setOnQueryEnteredListener(OnQueryEnteredListener listener) {
        this.queryListener = listener;
    }

    private View buildView() {

        View v = LayoutInflater.from(getContext())
                .inflate(R.layout.dialogfragment_enter_query, null, false);
        editQuery = v.findViewById(R.id.dialogfragment_isbn_edit);
        return v;
    }
}

File: app/src/main/java/at/shockbytes/dante/ui/fragment/dialog/QueryDialogFragment.kt
package at.shockbytes.dante.ui.fragment.dialog

import android.app.Dialog
import android.os.Bundle
import androidx.fragment.app.DialogFragment
import androidx.appcompat.app.AlertDialog
import android.view.LayoutInflater
import android.view.View
import android.view.Window
import android.view.WindowManager.LayoutParams.SOFT_INPUT_STATE_VISIBLE
import android.widget.EditText
import at.shockbytes.dante.R
import at.shockbytes.dante.util.showKeyboard
import kotterknife.bindView

/**
 * Author:  Martin Macheiner
 * Date:    30.08.2016
 */
class QueryDialogFragment : DialogFragment() {

    private val editQuery: EditText by bindView(R.id.dialogfragment_isbn_edit)

    private var queryListener: ((String) -> Unit)? = null

    private val queryView: View
        get() = LayoutInflater.from(context)
                .inflate(R.layout.dialogfragment_enter_query, null, false)

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        return AlertDialog.Builder(requireContext())
            .setView(queryView)
            .setNegativeButton(android.R.string.cancel) { _, _ -> dismiss() }
            .setPositiveButton(android.R.string.search_go) { _, _ ->
                // Remove blanks with + so query works also for titles
                val query = editQuery.text.toString().replace(' ', '+')
                queryListener?.invoke(query)
            }
            .create()
            .also { it.requestWindowFeature(Window.FEATURE_NO_TITLE) }
            .also { it.window?.setSoftInputMode(SOFT_INPUT_STATE_VISIBLE) }
    }

    fun setOnQueryEnteredListener(listener: (String) -> Unit): QueryDialogFragment {
        this.queryListener = listener
        return this
    }

    override fun onResume() {
        super.onResume()
        showKeyboard(editQuery)
    }

    companion object {

        fun newInstance(): QueryDialogFragment = QueryDialogFragment()
    }
}
